server/controllers/ingestion-controller.js

```diff
@@ -0,0 +1,188 @@
+const itemsController = require("./items-controller");
+const storesController = require("./stores-controller");
+const merchantsController = require("./merchants-controller");
+const fs = require("fs");
+const path = require("path");
+const { csvToJson, writeJsonToCsv, arrayEqual } = require("./../utils");
+const { HTTP_INTERNALERROR } = require("./../constants");
+const {
+  MERCHANTS_INGESTION_CSV,
+  INGESTION_STATUS_DOCUMENT,
+} = require("./../config");
+const { merchantsCollection, ingestionCollection } = require("./../firestore");
+const cloudStorage = require("./../cloud-storage");
+
+const bucket = cloudStorage.bucket;
+
+// common function to ingest a single merchant given configurations
+
+// Pull merchants.csv and check for new merchants, send request to update
+exports.sync = async (req, res) => {
```

Comment:
are you planning to add tests here or in a separate PR?
+ suggestion: split this function and have separate functions to filter merchants to onboard and to onboard merchants

Reply:
No tests yet unfortunately, since we are pulling files from cloud bucket (live), we would need to interject mocked functions to serve local files first before we can mock this onboarding function.
+1 for the suggestion. Yup! The next PR addresses the difference in onboarding a New v.s. Update existing merchant. Actual updating and write to merchants/items/stores collection then happens as a separate function.